packages/app/src/vis-packs/core/compound/MappedCompoundMatrixVis.tsx

```diff
@@ -0,0 +1,60 @@
+import { MatrixVis } from '@h5web/lib';
+import type {
+  ArrayShape,
+  Dataset,
+  Primitive,
+  PrintableCompoundType,
+  PrintableType,
+} from '@h5web/shared';
+import { capitalize } from 'lodash';
+import ndarray from 'ndarray';
+import { createPortal } from 'react-dom';
+import shallow from 'zustand/shallow';
+
+import MatrixToolbar from '../matrix/MatrixToolbar';
+import { useMatrixConfig } from '../matrix/config';
+import { getCellWidth } from '../matrix/utils';
+import { compoundFormatter } from './utils';
+
+interface Props {
+  dataset: Dataset<ArrayShape, PrintableCompoundType>;
+  value: Primitive<PrintableType>[][];
+  toolbarContainer: HTMLDivElement | undefined;
+}
+
+function MappedCompoundMatrixVis(props: Props) {
+  const { value, dataset, toolbarContainer } = props;
+  const { fields } = dataset.type;
+
+  const fieldNames = Object.keys(fields).map(capitalize);
+  const cellWidth = Math.max(...Object.values(fields).map(getCellWidth));
```

Comment:
I modified slightly `getCellWidth` to be able to pass the dtype rather than the dataset to be able to use it for field dtypes.

This way, we can set the width to fit the longest dtype.

Reply:
You could go half-way by calling `getCellWidth` with the compound type and adding a case for that that loops through the fields and recurses:

```ts
getCellWidth(dataset.type)

function getCellWidth(type: PrintableType | PrintableCompoundType): number {
  ...
  if (type.class === DTypeClass.Compound) {
    return Math.max(...Object.values(type.fields).map(getCellWidth));
  }
  ...
}
```